### PR TITLE
INIT BlastoiseFurnace

### PR DESCRIPTION
**Pull request by FunkyFishie**
Blast Furnace plugin;
description:
this plugin will handle blast furnace in a BF world , wont pay coffer or foremen.

works for : gold, steel, mithril, adamant, rune.
requires: ice gloves/goldsmiths gloves and a coal bag

a new sleepuntil method was added to script.java for some functionality to work properly

DEMO:
https://youtu.be/iyAOluFEpy8

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/microbot/Script.java [hunk 1]
```diff
@@ -65,6 +67,18 @@ public boolean sleepUntil(BooleanSupplier awaitedCondition, int time) {
         return done;
     }
 
+    public boolean sleepUntil(BooleanSupplier awaitedCondition, BooleanSupplier resetCondition, int timeout) {
```
Comment (chsami) [inline runelite-client/src/main/java/net/runelite/client/plugins/microbot/Script.java:71]: Why do you add an extra sleepuntil here? Cant you just use thé existing one by combining the await and reset condition in one boolean supplier?

File: runelite-client/src/main/java/net/runelite/client/plugins/microbot/blastoisefurnace/BlastoiseFurnaceConfig.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    @ConfigItem(
+            keyName = "BF Directions",
+            name = "BlastoiseDirections",
+            description = "Directions",
+            position = 1,
+            section = "bFSettings"
+    )
+    default String BlastoiseDirections() {
+        return "must have Ice Gloves or smiths gloves (i) equiped\n+\ncoalbag, stamina potions and energy potions in bank\n\n| if doing gold bars; must have Goldsmiths Gauntlet and bank your coalbag |\n\n(makes 1.5m an hour minimum with steel and should cover expenses)\n\ncurrent version does not support foremen or coffer refill\n\nim working on that aswell as full native use of antiban\nEnjoy! :) ";
```
Comment (chsami) [inline runelite-client/src/main/java/net/runelite/client/plugins/microbot/blastoisefurnace/BlastoiseFurnaceConfig.java]: We have a configinformation where you can put the script information in.

File: runelite-client/src/main/java/net/runelite/client/plugins/microbot/blastoisefurnace/BlastoiseFurnaceOverlay.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
+package net.runelite.client.plugins.microbot.blastoisefurnace;
+//
```
Comment (chsami) [inline runelite-client/src/main/java/net/runelite/client/plugins/microbot/blastoisefurnace/BlastoiseFurnaceOverlay.java]: It seems that you decompiled these classes. Where do you get thé code from?

**Review by chsami — CHANGES_REQUESTED** [review]

**Comment by FunkyRhythm** [comment]
testing

**Review by chsami — APPROVED** [review]

**Pull request merged**